Extension/src/LanguageServer/cppBuildTaskProvider.ts

```diff
@@ -170,10 +170,13 @@ export class CppBuildTaskProvider implements TaskProvider {
         const compilerPathBase: string = path.basename(compilerPath);
         const isCl: boolean = compilerPathBase.toLowerCase() === "cl.exe";
         const isClang: boolean = !isCl && compilerPathBase.toLowerCase().includes("clang");
-        // Double-quote the command if it is not already double-quoted.
+        // Double-quote the command if needed.
         let resolvedcompilerPath: string = isCl ? compilerPathBase : compilerPath;
-        if (resolvedcompilerPath && !resolvedcompilerPath.startsWith("\"") && resolvedcompilerPath.includes(" ")) {
-            resolvedcompilerPath = "\"" + resolvedcompilerPath + "\"";
+        if (os.platform() === 'win32') {
+            resolvedcompilerPath = util.quoteArgumentWindows(resolvedcompilerPath);
```

Comment:
Would it be better to make the API be util.quoteArgument -- i.e. check the platform in the implementation?

Reply:
Yep, this has been updated

Reply:
Where's the update?